### PR TITLE
Add support for Linea blockchain

### PR DESCRIPTION
**Pull request by NIXBNT**
This PR lays the initial framework for Linea support
A number of tasks need addressing:
- [x] Current rpc does not support newFilter and perhaps none on Linea do!
- [ ] Top 5 protocols by TVL might require specific integrations for adequate coverage
- [x] Fees were not specified for uniswap v2 forks might need to refetch static pool data

NB: The bot does not yet run due to lack of newFilter support

### CODE REVIEW

File: fastlane_bot/helpers/txhelpers.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def _get_transaction_info(self) -> (int, int, int, int):
                 self.get_max_priority_fee_per_gas_alchemy()
                 * self.ConfigObj.DEFAULT_GAS_PRICE_OFFSET
             )
-            if self.ConfigObj.NETWORK in ["ethereum", "coinbase_base"]
+            if self.ConfigObj.NETWORK in ["ethereum", "coinbase_base"] #TODO
```
[the remark this answers is not in the document]
Comment (barakman) [reply fastlane_bot/helpers/txhelpers.py:72]: No need, entire file will be overridden by the one on the 'simplify-tx-helpers' branch.

File: fastlane_bot/helpers/txhelpers.py [hunk 2]
```diff
@@ -584,7 +584,7 @@ def build_tx(
             max_gas_price = 3
             max_priority_fee = 3
 
-        if self.ConfigObj.NETWORK in ["ethereum", "coinbase_base"]:
```
[the remark this answers is not in the document]
Comment (barakman) [reply fastlane_bot/helpers/txhelpers.py:442]: No need, entire file will be overridden by the one on the 'simplify-tx-helpers' branch.

File: fastlane_bot/helpers/txhelpers.py [hunk 3]
```diff
@@ -769,7 +769,7 @@ def approve_token_for_arb_contract(self, token_address: str, approval_amount: in
             transaction hash
         """
         current_gas_price = self.web3.eth.get_block("pending").get("baseFeePerGas")
-        max_priority = int(self.get_max_priority_fee_per_gas_alchemy()) if self.ConfigObj.NETWORK in ["ethereum", "coinbase_base"] else 0
```
[the remark this answers is not in the document]
Comment (barakman) [reply fastlane_bot/helpers/txhelpers.py:660]: No need, entire file will be overridden by the one on the 'simplify-tx-helpers' branch.

**Review by barakman — APPROVED** [review]
Approve, provided that reverting @NIXBNT's change is the right thing to do:
```
exchange_df['last_updated_block'] = 0
exchange_df['cid'] = None changes
```

File: fastlane_bot/config/provider.py [hunk 4]
```diff
@@ -107,7 +107,7 @@ def __init__(self, network: ConfigNetwork, **kwargs):
         self.LOCAL_ACCOUNT = self.w3.eth.account.from_key(ETH_PRIVATE_KEY_BE_CAREFUL)
 
 
-        if network.NETWORK in [N.NETWORK_BASE, N.NETWORK_ETHEREUM, N.NETWORK_FANTOM, N.NETWORK_MANTLE]:
+        if network.NETWORK in [N.NETWORK_BASE, N.NETWORK_ETHEREUM, N.NETWORK_FANTOM, N.NETWORK_MANTLE]:  #TODO add Linea
```
[the remark this answers is not in the document]
Comment (barakman) [reply fastlane_bot/config/provider.py]: @Lesigh-3100 : what about this one?

**Review by Lesigh-3100 — APPROVED** [review]

**Pull request closed without merging**